require.js

```diff
@@ -1,26 +1,25 @@
 (function() {
   'use strict';
 
-  var globals = typeof window === 'undefined' ? global : window;
+  var globals = new Function('return this')();
   if (typeof globals.require === 'function') return;
 
   var modules = {};
   var cache = {};
   var aliases = {};
-  var has = ({}).hasOwnProperty;
+  var has = {}.hasOwnProperty;
```

Comment:
If that caused syntax errors, `var has = { a: 1 }` would not work either. I am pretty familiar with IE8 (IE9 is fine on ES) bugs, this is not one of them.